tables/AlternateMans/BufferedColumnarFile.h

```diff
@@ -0,0 +1,307 @@
+#ifndef CASACORE_BUFFERED_COLUMNAR_FILE_H_
+#define CASACORE_BUFFERED_COLUMNAR_FILE_H_
+
+#include <cassert>
+#include <complex>
+#include <cstdint>
+#include <string>
+#include <vector>
+
+#include "BitPacking.h"
+#include "RowBasedFile.h"
+
+namespace casacore {
+
+/**
+ * Class that provides binary table I/O. It is rather low-level, and requires
+ * the user to keep track of the columns and their datatype. It uses a simple
+ * binary format where the columns are interleaved into the output file.
+ * Booleans are written with bit-packing.
+ *
+ * This class limits calls to the read/write functions by caching a block of
+ * data around the last accessed row. The size of the buffer is templated to
+ * make testing of the caching easier. Production code can normally use the
+ * alias BufferedColumnarFile, which uses a reasonable buffer size.
+ *
+ * This class aims to be as fast as possible for large data files that are read
+ * from or written to consecutively. It does not try to optimize random access,
+ * partial data access and access is not transactional.
+ *
+ * The class uses exceptions to handle any I/O errors.
+ */
+template <uint64_t BufferSize = 100 * 1024>
+class VarBufferedColumnarFile : private RowBasedFile {
+ public:
+  using RowBasedFile::AddRows;
+  using RowBasedFile::DeleteRow;
+  using RowBasedFile::Filename;
+  using RowBasedFile::IsOpen;
+  using RowBasedFile::NRows;
+  using RowBasedFile::ReadHeader;
+  using RowBasedFile::Stride;
+  using RowBasedFile::WriteHeader;
+
+  VarBufferedColumnarFile() noexcept = default;
+
+  VarBufferedColumnarFile(const VarBufferedColumnarFile& rhs) = delete;
+  VarBufferedColumnarFile(VarBufferedColumnarFile&& rhs) noexcept
+      : packed_buffer_(std::move(rhs.packed_buffer_)),
+        block_changed_(rhs.block_changed_),
+        active_block_(rhs.active_block_),
+        rows_per_block_(rhs.rows_per_block_),
+        block_buffer_(std::move(rhs.block_buffer_)) {
+    rhs.block_changed_ = false;
+    rhs.active_block_ = 0;
+    rhs.rows_per_block_ = 0;
+  }
+
+  ~VarBufferedColumnarFile() noexcept {
+    if (IsOpen()) {
+      if (block_changed_) {
+        const uint64_t start_row = active_block_ * rows_per_block_;
+        const size_t n_rows_to_write =
+            std::min(rows_per_block_, std::max(NRows(), start_row) - start_row);
+        Seek(start_row * Stride() + DataLocation(), SEEK_SET);
+        WriteData(block_buffer_.data(), n_rows_to_write * Stride());
+        block_changed_ = false;
+      }
+    }
+  }
+
+  VarBufferedColumnarFile& operator=(VarBufferedColumnarFile&& rhs) {
+    RowBasedFile::operator=(std::move(rhs));
```

Comment:
`Close()` is not virtual, so the `Close()` in `RowBasedFile::operator=` doesn't call `VarBufferedColumnarFile::Close()`.
```suggestion
    Close();
    RowBasedFile::operator=(std::move(rhs));
```